reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/GlobalPrecisionDeclarationReductionOpportunities.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.graphicsfuzz.reducer.reductionopportunities;
+
+import com.graphicsfuzz.common.ast.TranslationUnit;
+import com.graphicsfuzz.common.ast.decl.Declaration;
+import com.graphicsfuzz.common.ast.decl.PrecisionDeclaration;
+import com.graphicsfuzz.common.ast.visitors.VisitationDepth;
+import com.graphicsfuzz.common.transformreduce.ShaderJob;
+import com.graphicsfuzz.common.typing.Scope;
+import com.graphicsfuzz.common.util.ListConcat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+public class GlobalPrecisionDeclarationReductionOpportunities {
+
+  private static List<GlobalPrecisionDeclarationReductionOpportunity> findOpportunitiesForShader(
+      TranslationUnit tu,
+      ReducerContext context) {
+    final List<GlobalPrecisionDeclarationReductionOpportunity> opportunities = new ArrayList<>();
+    List<Declaration> globalDeclarations = tu.getTopLevelDeclarations();
+    final HashMap<String, Scope> precisionMap = new HashMap<String, Scope>();
```

Comment:
Why use a map from String to scope; why not just a set of strings?

The scope could be useful for handling precision declarations inside functions, but since we do not support that I would prefer to get rid of it and use a set instead.  Doing a TODO(issue-url) with a comment to explain that the code is special-cased for this scenario would be my preference over using a map that only maps keys to null.

The type on the LHS of the declaration should be Map (or Set, if you use a set), rather than HashMap (or HashSet); best to only use the specific type as an argument to 'new' if possible.